### PR TITLE
Support for file Tags

### LINKED ISSUE #1161: Support for Tags
**Is your feature request related to a problem? Please describe.**
I really like the Tag feature in the Apple Finder. It would improve my workflow quite a bit, and the default Windows Explorer doesn't have it. Image below:

**Describe the solution you'd like**
- The ability to add Tags to files (and folders) with custom names and custom colors
- Tag list should appear on the sidebar below the drives
- Adding/Removing Tags via right click -> last row of the menu -> circles with colors. When a tag is selected, it should appear with a checkmark.
- Tag color in front of file name

**Describe alternatives you've considered**
There's not many good alternatives to tags... E.g., I could make a Tag for ToDo so that I don't have to write everything in a text file which is pretty annoying.

**Additional context**
Image from the Mac Finder:
grafik
grafik

### LINKED ISSUE #4558: Crashes On Launching Second Time
**Explain the bug**
On Launching App second time, got a crash, on first launch just changed some settings and everything was working perfectly fine.

**To reproduce**
1. After first launch, Everything was working fine, I just changed theme to dark mode and exited.
2. But on next launch I was on a screen like in below screenshot and got a immediate crash
3. Logs are provided below

**Expected behavior**
App should start normally.

**Logfile**
[debug_fulltrust.txt](https://github.com/files-community/Files/files/6292588/debug_fulltrust.txt)
[debug.txt](https://github.com/files-community/Files/files/6292590/debug.txt)

```
2021-04-12 00:50:37.4360|ERROR|.ctor|The RPC server is unavailable. (Exception from HRESULT: 0x800706BA)|System.Exception: The RPC server is unavailable. (Exception from HRESULT: 0x800706BA)
   at System.Runtime.InteropServices.McgMarshal.ThrowOnExternalCallFailed(Int32, RuntimeTypeHandle) + 0x21
   at __Interop.ComCallHelpers.Call(__ComObject, RuntimeTypeHandle, Int32, Void*) + 0xbe
   at __Interop.ForwardComStubs.Stub_15[TThis](__ComObject, Int32) + 0x2e
   at Files!<BaseAddress>+0x1d4f9b8
   at System.Runtime.InteropServices.WindowsRuntime.IIterator_PrivateRCWAdapter`1.MoveNext() + 0x104
   at Files.UserControls.Widgets.RecentFiles.<PopulateRecentsList>d__17.MoveNext() + 0x1ec
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__7_0(Object) + 0x1e
   at System.Action`1.Invoke(T) + 0x28
   at System.Threading.WinRTSynchronizationContext.Invoker.InvokeCore() + 0x33
   at System.Runtime.InteropServices.McgMarshal.ThrowOnExternalCallFailed(Int32, RuntimeTypeHandle) + 0x21
   at __Interop.ComCallHelpers.Call(__ComObject, RuntimeTypeHandle, Int32) + 0xb8
   at __Interop.ForwardComStubs.Stub_19[TThis](__ComObject, Int32) + 0x24
   at Microsoft.AppCenter.Utils.ApplicationLifecycleHelper.<.ctor>b__17_1(Object, UnhandledErrorDetectedEventArgs) + 0x3d
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21
   at Microsoft.AppCenter.Utils.ApplicationLifecycleHelper.<.ctor>b__17_1(Object, UnhandledErrorDetectedEventArgs) + 0x75
   at System.EventHandler`1.Invoke(Object, TEventArgs) + 0x2e
   at __Interop.Intrinsics.HasThisCall__24[TArg0](Object, IntPtr, Object, TArg0) + 0x36
   at Files!<BaseAddress>+0x158d6e0
   ```

**Screenshots**
RAoWJsxhvq

**System Information**
 - OS Version: 19041.867
 - App version: 1.5

**Additional comment**
N/A

### PR DESCRIPTION
**Pull request by gave92**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #1161
- Closes #4558

**Details of Changes**
Add details of changes here.
- Added support for tagging files and folders

Features;
- You can tag files and folders with a color badge through the context menu
- You can find all files with a tag by searching for "tag:\<TAGCOLOR\>" in the search-box (e.g "tag:orange")

TODO:
- [optional] add a more user-friendly way to search for tags
- [optional] add a page in Settings to edit tag name & color and to add/remove tags

**Validation**
How did you test these changes?
- [x] Built and ran the app

**Screenshots (optional)**

### CODE REVIEW

File: Common/Common.csproj [hunk 1]
```diff
@@ -4,4 +4,8 @@
     <TargetFramework>netstandard2.0</TargetFramework>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="LiteDB" Version="4.1.4" />
```
Comment (yaichenbaum) [inline Common/Common.csproj:8]: Can we use sqlite for this?

Comment (gave92) [reply Common/Common.csproj:8]: I'd like to, but I had problems with concurrent access to the DB. I'll see if it's still an issue.

Comment (gave92) [reply Common/Common.csproj:8]: SQLite adds 4Mb of dlls to the fulltrust process (DBLite is 300Kb), crashes all the time and API is ugly. For reference code for SQLite is in 29ddce0. No SQLite!

**Comment by Tropix126** [comment]
Do you think that we could use the same base colors as the macOS finder tags just to be consistent?
image

**Comment by yaichenbaum** [comment]
@Tropix126 We're working on our own set of colors to make sure they are accessible across light, dark and high contrast themes.

**Comment by winston-de** [comment]
there seem to be two backplates (one is reveal) when hovering over tags in the context menu, is this intended?
image

**Comment by yaichenbaum** [comment]
@winston-de I agree that should be changed. It can also use some rounded corners/

**Comment by winston-de** [comment]
nit: the tag icon is slightly misaligned with sync status icon in tiles view
image

**Comment by gave92** [comment]
The alignment issue should be fixed 👍. Let me know if there's something else that needs to be done before merging. If possible I'd like to get this merged as it is and work on "cool UI things" (settings, tags section on sidebar..) in a separate PR.

**Comment by winston-de** [comment]
One small thing, it looks like the tag isn't preserved when copying a folder that has a tag to another directory. Works fine for files and when moving folders, just not when copying them.

**Review by winston-de — DISMISSED** [review]
Lgtm

**Comment by gave92** [comment]
> It looks like the tag isn't preserved when copying a folder that has a tag to another directory

Thanks @winston-de, should be fixed now.

**Review by winston-de — APPROVED** [review]

**Comment by yaichenbaum** [comment]
@gave92 can you resolve the merge conflicts?

**Review by winston-de — APPROVED** [review]

**Pull request merged**